rfcs/066-graph_pipeline/README.md

```diff
@@ -0,0 +1,244 @@
+
+## Loading data from external sources
+
+Where possible, data for source concept nodes will be loaded to the graph in full. This currently applies to data from Library of Congress Subject Headings (LCSH), Library of Congress Name Authority File (LCNAF), and Medical Subject Headings (MeSH). Unless the size of the data is prohibitive (for example, in the case of Wikidata), the slightly higher storage needs will be outweighed by advantages associated with minimising dependencies, resulting in simplified pipeline logic and updates. 
+
+If we were to only load source nodes linked to our concepts, we would need to fetch new data every time a previously unseen concept is added to the graph. This includes any label-derived concepts we want to match to LoC and MeSH. For example, let's say we add a new LoC source node, we would in turn have to fetch other nodes from LoC (for example, broader/narrower terms). Following this, we would need to add any Wikidata concepts associated with these. For these Wikidata entries, we would then need to check if there is a MeSH term associated to it. If the MeSH term is not already present in the graph, any parent terms which appear in the hierarchy would also need to be loaded. 
+
+Instead, if we load source nodes and edges from LoC and MeSH in full, we only need to add an edge between the new concept and the source it is connected to. Source node and edge extraction can happen independently, and their frequency can be scheduled around updates to LoC and MeSH. To make sure there are no dependencies between different sources, the data will be loaded in multiple passes, whereby nodes and within-vocabulary edges will be loaded first, followed by edges between different vocabularies.
+
+### Library of Congress
+
+`SourceConcept` nodes from LCSH, as well as within-LoC edges will be loaded in full from https://id.loc.gov/download/authorities/subjects.skosrdf.jsonld.gz. `SourceName` nodes from LCNAF (including additional fields for birth date, death date, and birth place), as well as within-LoC edges will be loaded in full from https://id.loc.gov/download/authorities/names.skosrdf.jsonld.gz.
+
+Given that location terms appear in both LCSH and LCNAF, there needs to be some pipeline logic to extract `SourceLocation` nodes from both (for instance, via their grouping into `MADS/RDF Geographic`). This is to avoid cross-links between `SourceConcept` and `SourceName` nodes via Wikidata and MeSH. It may be desirable to create new source nodes for other entities such as organisations in the future, so we can collect relevant metadata on these from Wikidata. For example, organisations are currently part of LCNAF and will therefore be `SourceName` nodes with empty birth/death dates. However, unlike geographic locations, they are not affected by any cross-linking issues and will thus be deprioritised in the first iteration of the graph pipeline.
```

Comment:
I think some examples would be useful here in grasping the specific differences around locations being discussed here.

Reply:
Good idea, I've updated the README with some examples.

Reply:
It's not entirely clear to me how creating SourceLocation nodes helps avoid cross-linking and ambiguity with Wikidata nodes? I think it's probably my reading that's the issue rather than the description though so not blocking for merge.